hack/verify-kustomize

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+echo "Verifying kustomize"
+diff=$(git status --porcelain=v1)
+if [[ -n "${diff}" ]]; then
+  echo "${diff}"
+  echo
+  echo "Please commit all changes before verifying"
+  exit 1
+fi
+make generate-kustomize
+echo
+diff=$(git status --porcelain=v1)
+if [[ -n "${diff}" ]]; then
+  echo "${diff}"
+  echo
+  echo "Please run make generate-kustomize to fix the issue(s)"
```

Comment:
I'm a bit confused here, why would we have to rerun generate-kustomize?

Reply:
I just ran a few tests and I have no idea what this script is trying to do. It doesn't make sense why it's telling us to make generate-kustomize when it makes the call for us. There's also a bunch of weird behavior where it creates the binary directory for us, so the diff always exists and it always throws this message. I'm inclined to believe that this is useless and I'll probably remove it.